server/src/Infrastructure/Persistence/Interceptors/AuditableEntityInterceptor.cs

```diff
@@ -1,24 +1,60 @@
-﻿using StudyZen.Domain.Interfaces;
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Diagnostics;
+using StudyZen.Domain.Interfaces;
 using StudyZen.Domain.ValueObjects;
 
 namespace StudyZen.Infrastructure.Persistence;
 
-public static class AuditableEntityInterceptor
+public class AuditableEntityInterceptor : SaveChangesInterceptor
 {
-    public static void SetCreateStamp(object instance)
+    public override ValueTask<InterceptionResult<int>> SavingChangesAsync(
+        DbContextEventData eventData,
+        InterceptionResult<int> result,
+        CancellationToken cancellationToken = default)
+    {
+        UpdateAuditFields(eventData.Context!);
```

Comment:
We should check for null. Could add null check in UpdateAuditFields instead of using !